components/server/src/oauth-server/oauth-controller.ts

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright (c) 2020 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { AuthCodeRepositoryDB } from '@gitpod/gitpod-db/lib/typeorm/auth-code-repository-db';
+import { UserDB } from '@gitpod/gitpod-db/lib/user-db';
+import { User } from "@gitpod/gitpod-protocol";
+import { log } from '@gitpod/gitpod-protocol/lib/util/logging';
+import { OAuthException, OAuthRequest, OAuthResponse } from "@jmondi/oauth2-server";
+import * as express from 'express';
+import { inject, injectable } from "inversify";
+import { Env } from "../env";
+import { createAuthorizationServer } from './oauth-authorization-server';
+import { localAppClientID } from "./db";
+
+@injectable()
+export class OAuthController {
+    @inject(Env) protected readonly env: Env;
+    @inject(UserDB) protected readonly userDb: UserDB;
+    @inject(AuthCodeRepositoryDB) protected readonly authCodeRepositoryDb: AuthCodeRepositoryDB;
+
+    get oauthRouter(): express.Router {
+        const router = express.Router();
+        if (this.env.enableOAuthServer) {
```

Comment:
nit (happy path on the left):

```suggestion
if (!this.env.enableOAuthServer) {
    log.warn('OAuth server disabled!');
    return router;
}
```